paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -982,6 +982,12 @@ public class CoreOptions implements Serializable {
                     .defaultValue(MemorySize.ofMebiBytes(10))
                     .withDescription("The threshold for read file async.");
 
+    public static final ConfigOption<Boolean> DATA_DIRECTORIES_IGNORE_CLEAN =
+            key("data-directories.ignore-clean")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("Whether to ignore clean data directories.");
```

Comment:
1. Please use key "snapshot.expire.clean-empty-directories" and default value "true".
2. Please make description more clearer, for example, you can use "Whether to try to clean empty directories when expiring snapshots. Note that trying to clean directories might throw exceptions in filesystem, but in most cases it won't cause problems."